internal/worker/controlsocket/worker.go

```diff
@@ -34,9 +36,53 @@ const (
 	// userCreator is the listed "creator" of metrics users in state.
 	// This user CANNOT be a local user (it must have a domain), otherwise the
 	// model addUser code will complain about the user not existing.
-	userCreator = "controller@juju"
+	userCreator = "juju-metrics"
 )
 
+// UserService is the interface for the user service.
+type UserService interface {
+	// AddUser will add a new user to the database and return the UUID of the
+	// user if successful. If no password is set in the incoming argument,
+	// the user will be added with an activation key.
+	// The following error types are possible from this function:
+	//   - usererrors.UserNameNotValid: When the username supplied is not valid.
+	//   - usererrors.AlreadyExists: If a user with the supplied name already exists.
+	//   - usererrors.CreatorUUIDNotFound: If a creator has been supplied for the user
+	//     and the creator does not exist.
+	//   - auth.ErrPasswordNotValid: If the password supplied is not valid.
+	AddUser(ctx context.Context, arg service.AddUserArg) (user.UUID, []byte, error)
```

Comment:
Let's add an slice of permissions to `AddUserArg`, and if not empty, create them in the same transaction.

Reply:
That would make my life easier, but it does make my argument that permissions and uses should be in the same domain.

Reply:
@hmlanigan ☝️

Reply:
See: https://github.com/juju/juju/pull/17067